lib/cartopy/mpl/gridliner.py

```diff
@@ -53,21 +53,11 @@
 )
 
 
-def _fix_lons(lons):
-    """
-    Fix the given longitudes into the range ``[-180, 180]``.
-
-    """
-    lons = np.array(lons, copy=False, ndmin=1)
-    fixed_lons = ((lons + 180) % 360) - 180
-    # Make the positive 180s positive again.
-    fixed_lons[(fixed_lons == -180) & (lons > 0)] *= -1
-    return fixed_lons
-
-
 def _lon_hemisphere(longitude):
     """Return the hemisphere (E, W or '' for 0) for the given longitude."""
-    longitude = _fix_lons(longitude)
+    if longitude != 180:
+        # Wrap the longitude to the range -180 to 180, keeping positive 180s
+        longitude = ((longitude + 180) % 360) - 180
```

Comment:
There's a minor difference here in that `180 + 360 *n` was formerly considered +180, whereas now it'll be -180.

Reply:
True, I was ignoring the larger multiples. I think we need an intermediate variable to handle that as before, I just pushed up a new change.

Reply:
Thinking more generally here... this is only in the W/E labels function. Do we actually want a label on the 180s, or should we just strip that off altogether like we do for 0 already? Remove the E/W ambiguity.

Reply:
Yea, that might be something to do instead. I feel like there are probably differing opinions on which one to use, too.

Reply:
I agree.

Reply:
OK, there was one image failure with that update, so I pushed the change to remove the E/W label as that simplifies the logic a bit. I'm happy to remove that and go back to the previous logic if preferred too.

I looked up a few examples and it seems there are versions with and without the labels on 180, so no standard I could see at a quick glance.

Matlab does produce E/W labels on 180: https://www.mathworks.com/help/map/the-map-grid.html
Images without the E/W labels on 180: https://manoa.hawaii.edu/exploringourfluidearth/physical/world-ocean/locating-points-globe

Wikipedia seems to indicate you should label both the prime and antimeridian with either E or W: https://en.wikipedia.org/wiki/180th_meridian
> The [longitude](https://en.wikipedia.org/wiki/Longitude) at this line can be given as either [east](https://en.wikipedia.org/wiki/East) or [west](https://en.wikipedia.org/wiki/West).

Reply:
I was too undecided. I now went to the old version and kept the E/W labels as they were so we aren't changing anything anymore. This is the simple change for now to get ready for numpy 2.0 and we can re-evaluate the label naming in a follow-up PR later.

Reply:
I think we'd eventually want to go for something "smarter", that adds the direction if there are (potential for) two 180 labels. But this can be a separate PR.